ui_curses.c

```diff
@@ -1964,7 +1968,9 @@ static void update(void)
 	}
 
 	/* total time changed? */
-	if (play_library) {
+	if (cmus_queue_active()) {
+		needs_status_update += queue_needs_redraw();
```

Comment:
You need to clear `changed` to 0 when you read it similar to what's done for lib_editable, otherwise it'll still be set on every loop.

Reply:
There is the same problem with the playlist view then. `pl_needs_redraw` checks on `pl_editable_shared.win->changed`, but `changed` is only set to 0 after `pl_draw` is called. So that means the status bar redraws on every loop if playing from the playlist but `pl_draw` is not called, eg. if on another view.

Reply:
I was thinking to set all `changed` flags to 0 in `post_update`. Would that make sense? I will propose it for the play queue, and later it could be also extended to the playlists and library.

Reply:
I did a check, and it wouldn't work resetting the `changed` flag in `post_update()`. 
I would get a frozen queue while on the queue view, for example after typing `:clear`.
I am not sure why. 

Instead I clear the flag at the end of the `update()` function and that seems to work.

Reply:
True. I hadn't noticed that about playlist updates, but I guess you can just ignore it if you want.

> I would get a frozen queue while on the queue view

The queue view shouldn't freeze since it doesn't use `changed` as far as I see. But you probably meant playlist view since pl_draw() does check `changed` (although this optimization is probably not worth the headache).